parquet-hadoop/src/main/java/org/apache/parquet/hadoop/InternalParquetRecordReader.java

```diff
@@ -181,7 +181,7 @@ public void initialize(ParquetFileReader reader, ParquetReadOptions options) {
     this.columnCount = requestedSchema.getPaths().size();
     // Setting the projection schema before running any filtering (e.g. getting filtered record count)
     // because projection impacts filtering
-    reader.setRequestedSchema(requestedSchema);
+    this.requestedSchema = reader.setRequestedSchema(requestedSchema);
```

Comment:
`this.requestedSchema` is set in the line 180; so this line overrides the class field setting. Maybe you can use a temp local variable (with appropriate name) in the line 180

Reply:
Fixed. Thanks!